spring-integration/spring-boot-autoconfigure/src/main/java/ai/timefold/solver/spring/boot/autoconfigure/util/PojoInliner.java

```diff
@@ -0,0 +1,419 @@
+    }
+
+    public record PojoField(Class<?> type, String name, Object value) {
+    }
+
+    public static PojoField field(Class<?> type, String name, Object value) {
+        return new PojoField(type, name, value);
+    }
+
+    public static void inlineFields(TypeSpec.Builder typeBuilder, PojoField... fields) {
+        PojoInliner inliner = new PojoInliner();
+        for (PojoField field : fields) {
+            typeBuilder.addField(field.type(), field.name,
+                    javax.lang.model.element.Modifier.PRIVATE,
+                    javax.lang.model.element.Modifier.STATIC,
+                    javax.lang.model.element.Modifier.FINAL);
+        }
+        for (PojoField field : fields) {
+            inliner.inlineField(field.name(), field.value());
+        }
+        inliner.initializerBuilder.add("\n");
+        typeBuilder.addStaticBlock(inliner.initializerBuilder.build());
+    }
+
+    void inlineField(String fieldName, Object fieldValue) {
+        initializerBuilder.add("\n$L = $L;", fieldName, getInlinedPojo(fieldValue));
+    }
+
+    /**
+     * Serializes a Pojo to code that uses its no-args constructor
+     * and setters to create the object.
+     *
+     * @param pojo The object to be serialized.
+     * @return A string that can be used in a {@link CodeBlock.Builder} to access the object
+     */
+    String getInlinedPojo(Object pojo) {
+        // First, check for primitives
+        if (pojo == null) {
+            return "null";
+        }
+        if (pojo instanceof Boolean value) {
+            return value.toString();
+        }
+        if (pojo instanceof Byte value) {
+            // Cast to byte
+            return "((byte) " + value + ")";
+        }
+        if (pojo instanceof Character value) {
+            return "'\\u" + Integer.toHexString(value | 0x10000).substring(1) + "'";
+        }
+        if (pojo instanceof Short value) {
+            // Cast to short
+            return "((short) " + value + ")";
+        }
+        if (pojo instanceof Integer value) {
+            return value.toString();
+        }
+        if (pojo instanceof Long value) {
+            // Add long suffix to number string
+            return value + "L";
+        }
+        if (pojo instanceof Float value) {
+            // Add float suffix to number string
+            return value + "f";
+        }
+        if (pojo instanceof Double value) {
+            // Add double suffix to number string
+            return value + "d";
+        }
+
+        // Check for builtin classes
+        if (pojo instanceof String value) {
+            return "\"" + StringEscapeUtils.escapeJava(value) + "\"";
+        }
+        if (pojo instanceof Class<?> value) {
+            if (!Modifier.isPublic(value.getModifiers())) {
+                throw new IllegalArgumentException("Cannot serialize (" + value + ") because it is not a public class.");
```

Comment:
```suggestion
                throw new IllegalArgumentException("Cannot serialize (%s) because it is not a public class."
                        .formatted(value));
```

Let's consistently apply this convention in new code.
It makes for messages that are easier to read, especially as the number of arguments grows.